packages/cli/config/jest.js

```diff
@@ -62,45 +62,34 @@ function getRoleConfig(role) {
 async function getProjectConfig(targetPath, extraConfig) {
   const configJsPath = path.resolve(targetPath, 'jest.config.js');
   const configTsPath = path.resolve(targetPath, 'jest.config.ts');
-  // If the package has it's own jest config, we use that instead.
+  // If the package has it's own Jest config, we use that instead.
   if (await fs.pathExists(configJsPath)) {
     return require(configJsPath);
   } else if (await fs.pathExists(configTsPath)) {
     return require(configTsPath);
   }
 
-  // We read all "jest" config fields in package.json files all the way to the filesystem root.
-  // All configs are merged together to create the final config, with longer paths taking precedence.
-  // The merging of the configs is shallow, meaning e.g. all transforms are replaced if new ones are defined.
+  // We read the "jest" config fields from the current path 'package.json' file.
   const pkgJsonConfigs = [];
   let closestPkgJson = undefined;
-  let currentPath = targetPath;
 
   // Some confidence check to avoid infinite loop
-  for (let i = 0; i < 100; i++) {
-    const packagePath = path.resolve(currentPath, 'package.json');
-    const exists = fs.pathExistsSync(packagePath);
-    if (exists) {
-      try {
-        const data = fs.readJsonSync(packagePath);
-        if (!closestPkgJson) {
-          closestPkgJson = data;
-        }
-        if (data.jest) {
-          pkgJsonConfigs.unshift(data.jest);
-        }
-      } catch (error) {
-        throw new Error(
-          `Failed to parse package.json file reading jest configs, ${error}`,
-        );
+  const packagePath = path.resolve(targetPath, 'package.json');
```

Comment:
I think it's alright to remove this recursive ancestor config lookup, which will be a breaking change but I expect with minimal impact. I think we do need to keep the logic of loading in config from the root `package.json` if we're testing this package in isolation.

To clarify a bit, tests can be run in two main ways, either with `backstage-cli repo test`, which will use the multi-project config that we set up below. Or you can use a monorepo runner like Lerna and execute tests with `lerna run test`, this will run `backstagbe-cli package test` independently in each package, and with that setup we would want to merge any config in the project root into each package config.